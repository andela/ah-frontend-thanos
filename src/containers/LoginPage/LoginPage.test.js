import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Login from '../../components/Login';
import LoginPage from './index';

const mockStore = configureMockStore([thunk]);
let store;

const props = {
  signUp: jest.fn(),
  signupData: {
    errors: {
      password: ['Wrong password'],
    },
  },
};

describe('<Login />', () => {
  test('renders the component', () => {
    const LoginComponent = shallow(<Login onChange={jest.fn()} onSubmit={jest.fn()} />);
    expect(LoginComponent).toMatchSnapshot();
  });
});

describe('tesing login container', () => {
  let wrapper;
  beforeEach(() => {
    store = mockStore({
      LoginReducer: {
        errors: {
          email: ['The email not valid'],
          password: ['Wrong password'],
        },
      },
    });
  });
  it('should mount view all component', () => {
    wrapper = mount(
      <MemoryRouter>
        <Provider store={store}>
          <LoginPage props={props} loginUser={jest.fn} />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper.find('Login').length).toBe(1);
  });
});

describe('handle Invoke email for password reset', () => {
  let component;
  store = mockStore({});
  beforeEach(() => {
    component = mount(
      <MemoryRouter>
        <Provider store={store}>
          <LoginPage />
        </Provider>
      </MemoryRouter>,
    );
  });

  it('should submit user email and password without fail', () => {
    component
      .find('input#email')
      .simulate('change', { target: { name: 'email', value: 'ewrfg' } });
    component
      .find('input#email')
      .simulate('change', { target: { name: 'email', value: '' } });
    component
      .find('input#password')
      .simulate('change', { target: { name: 'password', value: '' } });
    component
      .find('input#password')
      .simulate('change', { target: { name: 'password', value: 'sdfg@dfg' } });

    component.find('form').simulate('submit');
  });
});
