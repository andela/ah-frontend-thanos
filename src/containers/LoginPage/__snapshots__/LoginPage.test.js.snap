// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders the component 1`] = `
<div
  className="container"
>
  <div
    className="col-left"
  >
    <img
      alt="Sign up poster"
      src="https://iviidev.info/downloads/images/reading.jpg"
    />
  </div>
  <div
    className="col-right"
  >
    <form
      onSubmit={[MockFunction]}
    >
      <h3
        className="text-center"
      >
        Sign In
      </h3>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="email"
          name="email"
          onChange={[MockFunction]}
          pattern="^[a-zA-Z0-9.!#$%&â€™*+/=?^_\`{|}~-]+@[a-zA-Z0-9-]+(?:.[a-zA-Z0-9-]+)*$"
          placeholder="Email"
          required={true}
          title="Please provide a valid email address"
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[MockFunction]}
          placeholder="**********"
          required={true}
          type="password"
        />
      </div>
      <div
        className="signin-buttons"
      >
        <button
          className="btn btn-primary sign-in-bt"
          id="signInBt"
          name="signInBt"
          type="submit"
        >
          Sign In
        </button>
        <div
          className="social-login-icons"
        >
          <small
            className="text-muted p-1"
          >
            Or Sign In with Social Media:
             
          </small>
          <SocialMediaLinks />
        </div>
      </div>
      <LoginSignupRedirect
        link="signup"
        linkName="Sign Up"
        message="Not yet a member? "
      />
    </form>
  </div>
</div>
`;
